scvi/external/csi/model/_model.py

```diff
@@ -0,0 +1,322 @@
+            Return latent embedding of the cycle pass.
+        give_mean
+            Return the posterior mean instead of a sample from the posterior.
+        batch_size
+            Minibatch size for data loading into model. Defaults to `scvi.settings.batch_size`.
+        as_numpy
+            Return in numpy rather than torch format.
+
+        Returns
+        -------
+        Latent Embedding
+        """
+        # Check model and adata
+        self._check_if_trained(warn=False)
+        # TODO extend to check if adata setup is correct wrt training data
+        adata = self._validate_anndata(adata)
+        if indices is None:
+            indices = np.arange(adata.n_obs)
+        # Prediction
+        # Do not shuffle to retain order
+        tensors_fwd = self._make_data_loader(
+            adata=adata, indices=indices, batch_size=batch_size, shuffle=False
+        )
+        predicted = []
+        for tensors in tensors_fwd:
+            # Inference
+            inference_inputs = self.module._get_inference_input(tensors)
+            inference_outputs = self.module.inference(**inference_inputs)
+            if cycle:
+                selected_system = self.module.random_select_systems(tensors["system"])
+                generative_inputs = self.module._get_generative_input(
+                    tensors,
+                    inference_outputs,
+                    selected_system=selected_system,
+                )
+                generative_outputs = self.module.generative(
+                    **generative_inputs, x_x=False, x_y=True
+                )
+                inference_cycle_inputs = self.module._get_inference_cycle_input(
+                    tensors=tensors,
+                    generative_outputs=generative_outputs,
+                    selected_system=selected_system,
+                )
+                inference_outputs = self.module.inference(**inference_cycle_inputs)
+            if give_mean:
+                predicted += [inference_outputs["z_m"]]
+            else:
+                predicted += [inference_outputs["z"]]
+
+        predicted = torch.cat(predicted)
+
+        if as_numpy:
+            predicted = predicted.cpu().numpy()
+        return predicted
+
+    @classmethod
+    @setup_anndata_dsp.dedent
+    def setup_anndata(
+        cls,
+        adata: AnnData,
+        system_key: str,
+        layer: Optional[str] = None,
+        categorical_covariate_keys: Optional[list[str]] = None,
+        categorical_covariate_embed_keys: Optional[list[str]] = None,
+        continuous_covariate_keys: Optional[list[str]] = None,
+        covariate_categ_orders: Optional[dict] = None,
+        covariate_key_orders: Optional[dict] = None,
+        system_order: Optional[list[str]] = None,
+        **kwargs,
+    ) -> AnnData:
+        """Prepare adata for input to Model
+
+        Parameters
+        ----------
+        adata
+            Adata object - will be modified in place.
+        system_key
```

Comment:
Would it be possible to rename this to `batch_key` for consistency with our other models, but explain in the docstring what should be expected in this field?